hummingbot/hummingbot
# feat/allow fees in different tokens

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
In this PR we introduce the required changes for the client to accept having fees for a trade in different tokens (something that happens with Binance connector, that sometimes can charge fees in BNB and sometimes in the trade quote token). The changes include:
- Remove fee_amount and fee_token from the order complete events, since that information was not being used anywhere
- Change PNL calculation logic to convert fees in different tokens to the quote token using the RateOracle as price provider
- Modification of the start command to always start also the RateOracle, for it to regularly update prices


**Tests performed by the developer**:
Added new unit tests.
Tested manually by running a PMM strategy against Binance and verifying that PNL is calculated without issues.


**Test Summary**:
- Test PR across different connectors except for connectors with a known issue and low volume exchange.
- Wait for both to both get a fill on bid and ask side and check the history performance.
- Check the rate of the fees paid in base asset using rate oracle 
- Manually compute the total PnL after getting the converted value of the fees paid and confirming that it matches or is close enough to the displayed total PnL
- Enable the use of a third-party asset for fee payment, for example, BNB, and wait for orders to get a filling.
- Manually convert the fees paid that are currently not in quote value and get the total fees paid.
- Manually compute for the total PnL again and confirmed that the PnL matches the output in the performance.
- Check AMM connectors as well to make sure that fees and pnl are calculated properly.


**Tips for QA testing**:
Please check that all connectors can correctly process order complete events.
Verify PNL calculation.

This PR comes from https://github.com/CoinAlpha/hummingbot/pull/94
